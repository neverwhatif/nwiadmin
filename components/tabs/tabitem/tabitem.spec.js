import React from 'react';
import { shallow } from 'enzyme';

import TabItem from './index';

jest.mock('nwiadmin/components/link', () => 'Link');

describe('TabItem', () => {
    const component = (<TabItem path="lorem" label="Lorem" />);
    const activeComponent = (<TabItem path="lorem" label="Lorem" isActive />);
    const disabledComponent = (<TabItem path="lorem" label="Lorem" isDisabled />);

    it('should render', () => {
        expect(shallow(component)).toMatchSnapshot();
    });
    it('should render active', () => {
        expect(shallow(activeComponent).hasClass('rootActive')).toBe(true);
    });
    it('should render disabled', () => {
        expect(shallow(disabledComponent).hasClass('rootDisabled')).toBe(true);
    });
});
