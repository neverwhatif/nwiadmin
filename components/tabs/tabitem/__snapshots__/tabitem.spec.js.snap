// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabItem should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TabItem
    isActive={false}
    isDisabled={false}
    label="Lorem"
    path="lorem"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Lorem",
      "className": "root",
      "to": "lorem",
    },
    "ref": null,
    "rendered": "Lorem",
    "type": "Link",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Lorem",
        "className": "root",
        "to": "lorem",
      },
      "ref": null,
      "rendered": "Lorem",
      "type": "Link",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
